### PR TITLE
25 commands

### LINKED ISSUE #25: Create command system

### PR DESCRIPTION
**Pull request by antoineverin**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,17 +13,20 @@ endif
...
 SRC         :=	main.cpp \
 	Client.cpp \
 	ClientSocket.cpp \
+	CommandRegistry.cpp \
 	Server.cpp \
 	ServerSocket.cpp \
 	Socket.cpp \
-	SocketObserver.cpp
+	SocketObserver.cpp \
+	commands/PingCommand.cpp
```
Comment (Neffi42) [inline Makefile]: It is better to create an "add prefix list with $CMDS_DIR variable" for the futures commands

File: src/CommandRegistry.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+									  Command* command) {
+	if (commands.find(name) != commands.end())
+		throw DuplicateException();
+	commands.insert(
+		std::pair<std::string, Command*>(strtolower(name), command));
+}
+
+Command* CommandRegistry::getCommand(const std::string& name) const {
+	std::map<std::string, Command*>::const_iterator cmd =
+		commands.find(strtolower(name));
```
Comment (Neffi42) [inline src/CommandRegistry.cpp:32]: Are all Irc servers putting everything to lower case ?

**Review by Neffi42 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (croueAlec) [reply Makefile]: we tried and did not succeed
we figured we'd let you show us how you'd approach the issue

**Review by croueAlec — APPROVED** [review]

File: src/CommandRegistry.cpp (hunk 2 above)
Comment (Neffi42) [reply src/CommandRegistry.cpp:32]: ok done

File: Makefile (hunk 1 above)
Comment (Neffi42) [reply Makefile]: ok done

**Review by Neffi42 — APPROVED** [review]

**Pull request merged**